### PR TITLE
docs(data-fetching): fix and simplify cookie proxy example

### PR DESCRIPTION
**Pull request by modex98**
cookies have this shape ==> `Set-Cookie: <cookie-name>=<cookie-value>; Domain=<domain-value>; Secure; `

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/2.guide/2.features/5.data-fetching.md [hunk 1]
```diff
@@ -216,7 +216,7 @@ export const fetchWithCookie = async (url: string, cookieName: string) => {
   const response = await $fetch.raw(url)
   if (process.server) {
     const cookies = Object.fromEntries(
-      response.headers.get('set-cookie')?.split(',').map((a) => a.split('='))
+      response.headers.get('set-cookie')?.split('; ').map((a) => a.split('='))
```
[the remark this answers is not in the document]
Comment (modex98) [reply docs/content/2.guide/2.features/5.data-fetching.md:219]: maybe this helps
```
export const fetchWithCookie = async (url: string, cookieName: string) => {
    const response = await $fetch.raw(url)

    if (process.server) {

        let cookies = response.headers.get('set-cookie')?.split(',');

        if(cookies) {
            for(let cookie of cookies) {
                if(cookie.startsWith(cookieName)) {
                    useCookie(cookieName).value = cookie.replace(cookieName, '');
                }
            }
        }
    }

    return response._data
}
```

[the remark this answers is not in the document]
Comment (pi0) [reply docs/content/2.guide/2.features/5.data-fetching.md:219]: We might use parseCookie from `cookie-es` for more realistic example.

[the remark this answers is not in the document]
Comment (modex98) [reply docs/content/2.guide/2.features/5.data-fetching.md:219]: hopfully this work)
```
export const fetchWithCookie = async (url: string, cookieName: string) => {
    const response = await $fetch.raw(url)

    if (process.server) {

        let cookies = response.headers.get('set-cookie')?.split(',');

        if(cookies) {
            for(let cookie of cookies) {
                if(cookie.startsWith(cookieName)) {
                    let payload = parse(cookie)
                    let _value = payload[cookieName]
                    delete payload[cookieName]
                    useCookie(cookieName, payload).value = _value
                    break;
                }
            }
        }
    }

    return response._data
}

```

**Review by pi0 — APPROVED** [review]
Thanks for PR. I've finally decided to keep example simple using split only. for loop can be tweaked for more advanced usage such as filtering cookie by name to proxy.

**Pull request merged**